.github/workflows/pr-build-image.yaml

```diff
@@ -55,17 +55,26 @@ jobs:
       - name: Check if Image Already Exists
         id: image-check
         run: |
-          IMAGE_TAG="pr-${{ github.event.number }}"
-          IMAGE_NAME="${{ env.REGISTRY }}/rhdh-community/rhdh:${IMAGE_TAG}"
+          IMAGE_TAG_PR="pr-${{ github.event.number }}"
+          IMAGE_TAG_COMMIT="pr-${{ github.event.number }}-${{ env.SHORT_SHA }}"
 
-          # Check if any image tag exists for the PR
-          IMAGE_TAGS=$(curl -s "https://quay.io/api/v1/repository/rhdh-community/rhdh/tag/" | jq -r --arg tag "$IMAGE_TAG" '.tags[] | select(.name | startswith($tag)) | .name')
+          IMAGE_NAME_PR="${{ env.REGISTRY }}/rhdh-community/rhdh:${IMAGE_TAG_PR}"
+          IMAGE_NAME_COMMIT="${{ env.REGISTRY }}/rhdh-community/rhdh:${IMAGE_TAG_COMMIT}"
 
-          if [ -n "$IMAGE_TAGS" ]; then
-            echo "Image $IMAGE_NAME already exists."
+          # Check if any image exists for the PR
+          IMAGE_EXISTS_PR=$(curl -s "https://quay.io/api/v1/repository/rhdh-community/rhdh/tag/" | jq -r --arg tag "$IMAGE_TAG_PR" '.tags[] | select(.name == $tag) | .name')
+
+          # Check if any image exists for the specific commit
+          IMAGE_EXISTS_COMMIT=$(curl -s "https://quay.io/api/v1/repository/rhdh-community/rhdh/tag/" | jq -r --arg tag "$IMAGE_TAG_COMMIT" '.tags[] | select(.name == $tag) | .name')
+
+          if [ -n "$IMAGE_EXISTS_COMMIT" ]; then
+            echo "Image $IMAGE_NAME_COMMIT already exists for the current commit."
             echo "image_exists=true" >> $GITHUB_ENV
+          elif [ -n "$IMAGE_EXISTS_PR" ]; then
```

Comment:
why check for the floating PR tag, if we need only the `IMAGE_TAG_COMMIT` to exist and therefore `IMAGE_EXISTS_COMMIT` to be true?

Reply:
Hi @nickboldt, great point! Checking only IMAGE_TAG_COMMIT simplifies the logic and ensures we only focus on commit-specific images. I've updated the workflow to remove the redundant check for the floating PR tag (IMAGE_TAG_PR). Let me know if you have any additional thoughts on this!